synapseclient/core/utils.py

```diff
@@ -342,12 +344,38 @@ def is_same_base_url(url1: str, url2: str) -> bool:
 def is_synapse_id_str(obj):
```

Comment:
Nit: I haven't looked at what other functions use this and I know you didn't add this function, but just wanted to point out that it is a bit misleading that the function doesn't return a boolean.  Do we know how other functions are using this?

When modifying other functions, can we add type hints?

Reply:
It's used across `activity.py`, `client.py`, `sync.py` and `conftest.py` but mostly in `client.py`. In the examples I've seen, the `return m.group(1)` return is never used. It's just used for `if`/`if not` statements to continue a process. I'll look through all the references to confirm this, but I'm pretty sure I can return a boolean without breaking code.

Yep, I'll update with type hints.

Reply:
3 unit tests fail when `is_synapse_id_str` returns `True` instead of the string. I'm going to leave this out for now since it's out of scope, and move forward updating the function with type hints, but me know if you think this should be in this PR